pysmt/logics.py

```diff
@@ -602,8 +602,13 @@ def __hash__(self):
 PYSMT_LOGICS = frozenset([QF_BOOL, QF_IDL, QF_LIA, QF_LRA, QF_RDL, QF_UF, QF_UFIDL,
                           QF_UFLIA, QF_UFLRA, QF_UFLIRA,
                           BOOL, LRA, LIA, UFLIRA, UFLRA,
-                          QF_BV, QF_UFBV])
+                          QF_BV, QF_UFBV,
+                          QF_ABV, QF_AUFBV, QF_AUFLIA, QF_ALIA
+                          #MG: QF_AX?
```

Comment:
No, it's not a SAT problem. You can have uninterpreted sorts for both the index and the element. QF_AX contains SMT formulas only the theory of arrays only. Example:

```
(declare-sort Index 0)
(declare-sort Element 0)
(declare-fun a () (Array Index Element))
(declare-fun b () (Array Index Element))
(declare-fun i () Index)
(declare-fun e () Element)
(declare-fun e2 () Element)
(assert (= (store a i e) b))
(assert (not (= (select a i) e)))
(assert (= a b))
(check-sat)
```